testing/regress/ecl/fileservice.ecl

```diff
@@ -33,5 +33,10 @@ SEQUENTIAL(
   File.RenameLogicalFile(prefix + 'afterrename1.d00', prefix + 'scope1::scope2::afterrename2.d00'),
   File.RenameLogicalFile(prefix + 'scope1::scope2::afterrename2.d00', prefix + 'scope1::afterrename3.d00'),
   OUTPUT(DATASET(prefix + 'scope1::afterrename3.d00', rec, FLAT)),
-  File.DeleteLogicalFile(prefix + 'scope1::afterrename3.d00')
+  File.DeleteLogicalFile(prefix + 'scope1::afterrename3.d00'),
+  OUTPUT(ds, , prefix + 'renametest.d00', OVERWRITE),
+  OUTPUT(ds, , prefix + 'afterrename1.d00', OVERWRITE),
+  // Remane with overwrite allowed
+  File.RenameLogicalFile(prefix + 'renametest.d00', prefix + 'afterrename1.d00', true),
```

Comment:
I think there may be some untested problems.
1) Rename external or foreign with overwrite.
Both are disallowed and will error on rename, but the delete will have already happened.. 
Foreign delete should fail, so may be ok.
External delete .. I'm not sure, worth testing.
i.e. need to make sure it doesn't succeed to delete - but then fail on rename.
2) Rename in a transaction.
Actually fileservices.ecl doesn't have any transaction tests at the moment.
I think there may be problems with rename in particular, related to the way delete happens.
See addDelayedDelete/CDelayedDelete in dadfs.cpp

Reply:
As we discussed, I will test this delete, rename in transaction.